package/MDAnalysis/coordinates/chemfiles.py

```diff
@@ -0,0 +1,333 @@
+
+    def write(self, obj):
+        """Write object `obj` at current trajectory frame to file.
+
+        Topology for the output is taken from the `obj` or default
+        to the value of the `topology` keyword supplied to the
+        :class:`ChemfilesWriter` constructor.
+
+        Parameters
+        ----------
+        obj : Universe or AtomGroup
+            The :class:`~MDAnalysis.core.groups.AtomGroup` or
+            :class:`~MDAnalysis.core.universe.Universe` to write.
+        """
+        if hasattr(obj, "atoms"):
+            if hasattr(obj, 'universe'):
+                # For AtomGroup and children (Residue, ResidueGroup, Segment)
+                ts_full = obj.universe.trajectory.ts
+                if ts_full.n_atoms == obj.atoms.n_atoms:
+                    ts = ts_full
+                else:
+                    # Only populate a time step with the selected atoms.
+                    ts = ts_full.copy_slice(atoms.indices)
+            elif hasattr(obj, 'trajectory'):
+                # For Universe only --- get everything
+                ts = obj.trajectory.ts
+        else:
+            if isinstance(obj, base.Timestep):
+                ts = obj
+                topology = None
+            else:
+                raise TypeError("No Timestep found in obj argument")
+
+        frame = self._timestep_to_chemfiles(ts)
+        frame.topology = self._topology_to_chemfiles(obj, len(frame.atoms))
+        self._file.write(frame)
+
+    def write_next_timestep(self, ts):
+        """Write timestep object into trajectory.
+
+        Parameters
+        ----------
+        ts: TimeStep
+        """
+        frame = self._timestep_to_chemfiles(ts)
+        self._file.write(frame)
+
+    def _timestep_to_chemfiles(self, ts):
+        """
+        Convert a Timestep to a chemfiles Frame
+        """
+        frame = Frame()
+        frame.resize(ts.n_atoms)
+        if ts.has_positions:
+            frame.positions[:] = ts.positions[:]
+        if ts.has_velocities:
+            frame.add_velocities()
+            frame.velocities[:] = ts.velocities[:]
+        frame.cell = UnitCell(*ts.dimensions)
+        return frame
+
+    def _topology_to_chemfiles(self, obj, n_atoms):
+        """
+        Convert an AtomGroup or an Universe to a chemfiles Topology
+        """
+        topology = Topology()
+        if not hasattr(obj, "atoms"):
+            # use an empty topology
+            topology.resize(n_atoms)
+            return topology
+
+        # (1) add all atoms to the topology
+        residues = {}
+        for atom in obj.atoms:
+            name = getattr(atom, 'name', "")
+            type = getattr(atom, 'type', name)
+            chemfiles_atom = Atom(name, type)
+
+            if hasattr(atom, 'altLoc'):
+                chemfiles_atom["altloc"] = str(atom.altLoc)
+
+            if hasattr(atom, 'segid'):
+                # TODO: what is the type of atom.segid?
+                chemfiles_atom["segid"] = atom.segid
```

Comment:
Just to be sure, `atom.segid` should be an integer, right?

Reply:
In general, it's not an integer:
```python
>>> import MDAnalysis as mda
>>> from MDAnalysisTests.datafiles import GRO
>>> u = mda.Universe(GRO)
>>> print(u.atoms[0].segid)
SYSTEM
```

Reply:
If you need an integer, use `atom.segindex`

Reply:
Thanks!